lib/octicons_react/script/build.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const transform = require('babel-core').transform;
+const octicons = require('octicons');
+const fs = require('fs-extra');
+const path = require('path');
+const lodash = require('lodash')
+
+const BUILD_PATH = path.join(__dirname, '..', 'build');
+
+fs.emptyDirSync(BUILD_PATH)
+
+// Transform code to ES5
+const getTransformedSourceCode = (originalSource) => transform(originalSource, {
+  presets: [ 'env', 'react' ]
+}).code;
+
+/**
+ * Template: React components
+ */
+const getReactSource = ({ componentName, options, svgPaths }) => {
+  let m = []
+  while ((m = /\s([a-z]*\-[a-z]*)=/gi.exec(svgPaths)) != null) {
+    let mat = m.pop()
+    svgPaths = svgPaths.replace(mat, lodash.camelCase(mat))
+  }
+  return getTransformedSourceCode(`
+import createIconComponent from '../utils/createIconComponent';
+import React from 'react';
+const ${componentName} = createIconComponent({ content: <g>${svgPaths}</g>, options: ${options} });
+${componentName}.displayName = '${componentName}';
+export default ${componentName};
+`);
+}
+
+/**
+ * Template: createIconComponent
+ */
+const getCreateIconSource = () => getTransformedSourceCode(`
+import React from "react";
+
+const createIconComponent = ({ content, options }) => {
+
+  const prepare = props => {
+    const { ariaLabel, medium, large } = props;
+    const attributes = Object.assign({}, options)
+
+    // Default octicon sizes to small
+    attributes["height"] = 16;
+
+    // Read in small, medium, large props
+    if (medium) {
+      attributes["height"] = 32;
+    } else if (large) {
+      attributes["height"] = 64;
+    }
+
+    // Calculate the width based on height
+    attributes["width"] =
+      attributes["height"] * options["width"] / options["height"];
+
+    // If the user passed in aria-label
+    if (ariaLabel) {
+      attributes["aria-label"] = ariaLabel;
+      attributes["role"] = "img";
+
+      // Un-hide the icon
+      delete attributes["aria-hidden"];
+    }
+
+    return attributes;
+  };
+
+  const alignment = props => {
+    const { top, middle } = props;
+
+    if (top) {
+      return "text-top";
+    } else if (middle) {
+      return "middle";
+    }
+    return "text-bottom";
+  };
+
+  return (props) => {
+    const svgStyle = {
+      display: "inline-block",
+      fill: "currentColor",
+      verticalAlign: alignment(props),
+      userSelect: "none"
+    };
+
+    return (
+      <svg
+        {...prepare(props)}
+        style={svgStyle} >
+        {content}
+      </svg>
+    )
+  }
+}
+
+export default createIconComponent;
+`);
+
+/**
+ * Template: createIconComponent
+ */
+const getIconIndexSource = () => {
+  const icons = Object.values(octicons).map((octicon) => {
+    const { name } = octicon;
+    const componentName = `${lodash.upperFirst(lodash.camelCase(name))}Octicon`;
+    return `export const ${componentName} = require('./icons/${name}.js').default;
+    octicons['${name}'] = ${componentName};`
```

Comment:
So the presence of this internal `octicons` registery is going to bust tree shaking. Because mutating this object is a top level side effect, the compiler will be forced to always include all icons no matter.

Reply:
😢 

Ok, I've come to the realization that my dream of single imports isn't going to work. After talking through with the design-systems team, that what we want to default to is dynamic `name=` property. We feel like even though your use case is best practice, we want people to be able to experiment with the full library. 

Once users reach a point where they're shipping their code to production, then can switch to the optimized imports and only get specific icons.

```js
// Importing all the octicons
import Octicon from '@github/octicons-react'

// For optimizing your code and importing only specific octicons.
import {AlertOcticon, XOcticon} from '@github/octicons-react/named'
```

@josh @koddsson does this work for you?

Reply:
I don't mind the longer name for optimised imports but I'll let @josh have the final say 😸